flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -818,71 +800,46 @@ else if (numConsumers == 0) {
 			// ----------------------------------------------------------------
-			else {
-				if (consumerState == RUNNING) {
-					final LogicalSlot consumerSlot = consumer.getAssignedResource();
-
-					if (consumerSlot == null) {
-						// The consumer has been reset concurrently
-						continue;
-					}
-
-					final TaskManagerLocation partitionTaskManagerLocation = partition.getProducer()
-							.getCurrentAssignedResource().getTaskManagerLocation();
-					final ResourceID partitionTaskManager = partitionTaskManagerLocation.getResourceID();
-
-					final ResourceID consumerTaskManager = consumerSlot.getTaskManagerLocation().getResourceID();
-
-					final ResultPartitionID partitionId = new ResultPartitionID(partition.getPartitionId(), attemptId);
-
-					final ResultPartitionLocation partitionLocation;
+			else if (consumerState == DEPLOYING || consumerState == RUNNING) {
+				final LogicalSlot consumerSlot = consumer.getAssignedResource();
+				if (consumerSlot == null) {
+					// The consumer has been reset concurrently
+					continue;
+				}
 
-					if (consumerTaskManager.equals(partitionTaskManager)) {
-						// Consuming task is deployed to the same instance as the partition => local
-						partitionLocation = ResultPartitionLocation.createLocal();
-					}
-					else {
-						// Different instances => remote
-						final ConnectionID connectionId = new ConnectionID(
-								partitionTaskManagerLocation,
-								partition.getIntermediateResult().getConnectionIndex());
+				final TaskManagerLocation partitionTaskManagerLocation = partition.getProducer()
+					.getCurrentAssignedResource().getTaskManagerLocation();
+				final ResourceID partitionTaskManager = partitionTaskManagerLocation.getResourceID();
+				final ResourceID consumerTaskManager = consumerSlot.getTaskManagerLocation().getResourceID();
 
-						partitionLocation = ResultPartitionLocation.createRemote(connectionId);
-					}
+				final ResultPartitionLocation partitionLocation;
+				if (consumerTaskManager.equals(partitionTaskManager)) {
+					// Consuming task is deployed to the same instance as the partition => local
+					partitionLocation = ResultPartitionLocation.createLocal();
+				} else {
+					// Different instances => remote
+					final ConnectionID connectionId = new ConnectionID(
+						partitionTaskManagerLocation,
+						partition.getIntermediateResult().getConnectionIndex());
+					partitionLocation = ResultPartitionLocation.createRemote(connectionId);
+				}
 
-					final InputChannelDeploymentDescriptor descriptor = new InputChannelDeploymentDescriptor(
-							partitionId, partitionLocation);
+				final ResultPartitionID partitionId = new ResultPartitionID(partition.getPartitionId(), attemptId);
+				final InputChannelDeploymentDescriptor descriptor = new InputChannelDeploymentDescriptor(
+					partitionId, partitionLocation);
+				final PartitionInfo partitionInfo = new PartitionInfo(partition.getIntermediateResult().getId(), descriptor);
```

Comment:
all code, needed to create `PartitionInfo`, could be packed into e.g. `PartitionInfo .create(ExecutionEdge)` to simplify `scheduleOrUpdateConsumers`.